annotators/midas_predictor/model_training.py

```diff
@@ -0,0 +1,152 @@
+val, test = train_test_split(val_test, test_size=0.5, random_state=42)
+
+target_train = [sample["predict"] for sample in train]
+target_val = [sample["predict"] for sample in val]
+target_test = [sample["predict"] for sample in test]
+
+target_train = pd.json_normalize(target_train)
+target_val = pd.json_normalize(target_val)
+target_test = pd.json_normalize(target_test)
+
+print(set(target_val["midas"].value_counts().index.tolist()) - set(target_train["midas"].value_counts().index.tolist()))
+
+print(
+    set(target_test["midas"].value_counts().index.tolist()) - set(target_train["midas"].value_counts().index.tolist())
+)
+
+print(target_train["midas"].value_counts(normalize=True))
+
+print(target_val["midas"].value_counts(normalize=True))
+
+print(target_test["midas"].value_counts(normalize=True))
+
+with open("data/annotated/midas_train.json", "w", encoding="utf-8") as f:
+    json.dump(train, f, ensure_ascii=False, indent=2)
+
+with open("data/annotated/midas_val.json", "w", encoding="utf-8") as f:
+    json.dump(val, f, ensure_ascii=False, indent=2)
+
+with open("data/annotated/midas_test.json", "w", encoding="utf-8") as f:
+    json.dump(test, f, ensure_ascii=False, indent=2)
+
+Midas2Id = {
+    "appreciation": 0,
+    "command": 1,
+    "comment": 2,
+    "complaint": 3,
+    "dev_command": 4,
+    "neg_answer": 5,
+    "open_question_factual": 6,
+    "open_question_opinion": 7,
+    "opinion": 8,
+    "other_answers": 9,
+    "pos_answer": 10,
+    "statement": 11,
+    "yes_no_question": 12,
+}
+with open("data/annotated/midas_train.json", "r", encoding="utf8") as f:
+    train = json.load(f)
+
+with open("data/annotated/midas_val.json", "r", encoding="utf8") as f:
+    val = json.load(f)
+
+with open("data/annotated/midas_test.json", "r", encoding="utf8") as f:
+    test = json.load(f)
+
+print(len(train), len(val), len(test))
+
+
+TFHUB_CACHE_DIR = os.environ.get("TFHUB_CACHE_DIR", None)
+if TFHUB_CACHE_DIR is None:
+    os.environ["TFHUB_CACHE_DIR"] = "/root/tfhub_cache"
+
+USE_MODEL_PATH = os.environ.get("USE_MODEL_PATH", None)
+if USE_MODEL_PATH is None:
+    USE_MODEL_PATH = "https://tfhub.dev/google/universal-sentence-encoder/4"
+
+encoder = hub.load(USE_MODEL_PATH)
+
+midas_vectorizer = MidasVectorizer(
+    text_vectorizer=encoder, midas2id=Midas2Id, context_len=3, embed_dim=512  # USE  # USE vector size
+)
+
+midas_vectorizer.context_vector(train[0]["previous_text"], train[0]["midas_vectors"])
+
+train_dataloader = MidasDataset(data=train, vectorizer=midas_vectorizer, batch_size=len(train), shuffle=False)
+
+val_dataloader = MidasDataset(data=val, vectorizer=midas_vectorizer, batch_size=len(val), shuffle=False)
+
+test_dataloader = MidasDataset(data=test, vectorizer=midas_vectorizer, batch_size=len(test), shuffle=False)
+
+for X_train, y_train in train_dataloader:
+    break
+
+for X_val, y_val in val_dataloader:
+    break
+
+for X_test, y_test in test_dataloader:
```

Comment:
test split can be removed from here. We have it for toloka only.

Reply:
left only to see data sizes